ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/BatchedValidatingCommitter.scala

```diff
@@ -18,6 +18,37 @@ import com.daml.ledger.validator.batch.{
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
+/**
+  * Orchestrates committing to a ledger after validating submissions. Supports parallel validation.
+  * Example usage, assuming a [[com.daml.ledger.participant.state.kvutils.api.BatchingLedgerWriter]] sends the
+  * batched submissions over the wire:
+  * {{{
+  *   ...
+  *   private val ledgerStateOperations = ...
+  *   private val validator = BatchedSubmissionValidator.create(ledgerStateOperations)
+  *   private val validatingCommitter = BatchedValidatingCommitter(
+  *       () => Instant.now(),
+  *       validator)
+  *   ...
+  *
+  *   def commitRequestHandler(request: CommitRequest): Future[CommitResponse] =
+  *     validatingCommitter.commit(
+  *         request.correlationId,
+  *         request.envelope,
+  *         request.participantId,
+  *         ledgerStateOperations)
+  *       .map(...)
+  * }}}
+  *
+  * In case caching is enabled (i.e., [[stateValueCache]] is not a [[Cache.none]]) then for each
+  * request we cache the read state from the ledger and update the cache with the committed state.
+  *
+  * @param now function implementing resolution of current time when processing submission
+  * @param keySerializationStrategy instance to use for serializing & namespacing state keys
+  * @param validator validator instance to use
```

Comment:
```suggestion
  * @param validator
```

Reply:
@fabiotudone-da I find a `@param` tag without a description pretty useless. If there's nothing to document, then I'd completely omit the line here.

Reply:
Agreed regarding not adding a tag without meaningful descriptions. However, as not all arguments are trivial I'm leaning towards enumerating all of them rather than selectively describing some.